lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -483,10 +474,39 @@ export default class BitcoinProcessor {
    * value.
    */
   private getFirstBlockInGroup (block: number): number {
-    const groupId = this.getGroupId(block);
+    const groupId = this.getGroupIdFromBlock(block);
     return groupId * ProtocolParameters.groupSizeInBlocks;
   }
 
+  private async getStartingBlockForInitialization (): Promise<IBlockInfo | undefined> {
+    // We look at the latest data that we saved in the quantile calculator. The latest
+    // Id in there tells us which group of blocks have been processed.
+    const lastSavedGroupId = await this.quantileCalculator.getLastGroupId();
+
+    if (!lastSavedGroupId) {
+      // We are returning undefined because at the time of this refactor, the calling code
+      // is expecting an undefined for this case and having this code return the genesis
+      // block is going to cause more refactor.
+      return undefined;
+    }
+
+    const startingBlockForLastSavedGroupId = this.getStartingBlockFromGroupId(lastSavedGroupId);
+
+    // We want to delete the last group Id because we might not have saved all the transaction(s)
+    // in that group. So removing that ensures that we redo this block which will ensure that
+    // all the transactions are processed again.
+    await this.quantileCalculator.removeGroupsGreaterThanOrEqual(lastSavedGroupId);
+
+    // The starting block is included in the lastGroupId and since we are removing the lastGroupId
+    // we should remove that block from the  the transaction store as well
+    await this.transactionStore.removeTransactionsLaterThan(startingBlockForLastSavedGroupId - 1);
```

Comment:
`removeTransactionsLaterThan` takes in a transaction number, not a block number, so you'll need to construct a transaction number using `TransactionNumber.construct()`

Reply:
```suggestion
    await this.transactionStore.removeTransactionsLaterThan(TransactionNumber.construct(startingBlockForLastSavedGroupId - 1, 0));
```

Reply:
```suggestion
    await this.transactionStore.removeTransactionsLaterThan(TransactionNumber.construct(startingBlockForLastSavedGroupId, 0) - 1);
```

Reply:
Updated code to convert block to transaction number and remove it.